transaction/executer_test.go

```diff
@@ -233,363 +232,90 @@ func Test_Executer_TransactionIDGiven_NoReplay(t *testing.T) {
+		WantReplayInput interface{}
+	}{
+		{ // 0
+			TrialOutput:     []byte("hello world"),
+			WantReplayInput: "hello world",
+		},
+		{ // 1
+			TrialOutput:     float64(4.3),
+			WantReplayInput: "4.3",
+		},
+		{ // 2
+			TrialOutput:     nil,
+			WantReplayInput: nil,
+		},
+		{ // 3
+			TrialOutput:     "hello world",
+			WantReplayInput: "hello world",
+		},
+		{ // 4
+			TrialOutput:     "",
+			WantReplayInput: "",
+		},
+		{ // 5
+			TrialOutput: struct {
+				Foo string `json:"foo"`
+				Bar int    `json:"bar"`
+			}{
+				Foo: "foo-val",
+				Bar: 43,
+			},
+			WantReplayInput: `{"foo":"foo-val","bar":43}`,
+		},
+	}
+
+	for i, tc := range tests {
+		config := DefaultExecuterConfig()
+		newExecuter, err := NewExecuter(config)
 		if err != nil {
 			t.Fatal("expected", nil, "got", err)
 		}
 
-		if trialOutput != 4.3 {
-			t.Fatal("expected", 4.3, "got", trialOutput)
-		}
-		if replayInput != 0 {
-			t.Fatal("expected", 0, "got", replayInput)
-		}
-	}
+		var replayInput interface{}
 
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
-		if err != nil {
-			t.Fatal("expected", nil, "got", err)
-		}
-
-		if trialOutput != 4.3 {
-			t.Fatal("expected", 4.3, "got", trialOutput)
+		replay := func(context context.Context, v interface{}) error {
+			replayInput = v
+			return nil
 		}
-		if replayInput != 4.3 {
-			t.Fatal("expected", 4.3, "got", replayInput)
+		trial := func(context context.Context) (interface{}, error) {
+			return tc.TrialOutput, nil
 		}
-	}
-}
-
-func Test_Executer_TransactionResult_Nil(t *testing.T) {
-	config := DefaultExecuterConfig()
-	newExecuter, err := NewExecuter(config)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
 
-	var replayInput interface{}
+		var ctx context.Context
+		var executeConfig ExecuteConfig
+		{
+			ctx = context.Background()
+			ctx = transactionid.NewContext(ctx, "test-transaction-id")
 
-	replay := func(context context.Context, v interface{}) error {
-		replayInput = v
-		return nil
-	}
-	trial := func(context context.Context) (interface{}, error) {
-		return nil, nil
-	}
-
-	var ctx context.Context
-	var executeConfig ExecuteConfig
-	{
-		ctx = context.Background()
-		ctx = transactionid.NewContext(ctx, "test-transaction-id")
-
-		executeConfig = newExecuter.ExecuteConfig()
-		executeConfig.Replay = replay
-		executeConfig.Trial = trial
-		executeConfig.TrialID = "test-trial-ID"
-	}
-
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
-		if err != nil {
-			t.Fatal("expected", nil, "got", err)
-		}
-
-		if replayInput != nil {
-			t.Fatal("expected", nil, "got", replayInput)
+			executeConfig = newExecuter.ExecuteConfig()
+			executeConfig.Replay = replay
+			executeConfig.Trial = trial
+			executeConfig.TrialID = "test-trial-ID"
 		}
-	}
 
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
+		err = newExecuter.Execute(ctx, executeConfig)
 		if err != nil {
-			t.Fatal("expected", nil, "got", err)
+			t.Errorf("#%d: want nil, got %#v", i, err)
```

Comment:
I'd like to keep that. This is like it is in `etcd` codebase.
I think I earned that :D